Editor/Tests/Algorithm/AdaptiveScanAlgorithmTests.cs

```diff
@@ -18,17 +18,24 @@ private class ControlledValue : IControlled
             public double stepValue;
             public ControlledBasis ControlledBasis => ControlledBasis.FloatingPoint;
 
+            public int __VERSION__ => 1;
+            
             public void Deserialize(JsonObject data)
             {
                 throw new System.NotImplementedException();
             }
 
+            public bool TryUpgradeVersion(JsonObject serializedData)
+            {
+                return true;
```

Comment:
Could this do some helpful things for the base class? I think perhaps this should be a final override and provide a protected method for going from version N to N+1. It doesn't have to be called at all if the version is not present in the JsonObject, and otherwise it can loop through the necessary version upgrades.